tests/executors/dask_executor.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import time
+import unittest
+
+from airflow import configuration
+from airflow.models import DAG, DagBag, TaskInstance, State
+from airflow.jobs import BackfillJob
+from airflow.operators.python_operator import PythonOperator
+
+try:
+    from airflow.executors import DaskExecutor
+    from distributed import LocalCluster
+    DASK = True
+except ImportError:
+    DASK = False
+
+DEFAULT_DATE = datetime.datetime(2017, 1, 1)
+
+
+if DASK:
+
+    class DaskExecutorTest(unittest.TestCase):
+
+        def setUp(self):
+            self.dagbag = DagBag(include_examples=True)
+
+        @unittest.skipIf('sqlite' in configuration.get('core', 'sql_alchemy_conn'),
+                         "concurrent access not supported in sqlite")
+        def test_executor_fuctions(self):
+            cluster = LocalCluster(nanny=False)
+
+            executor = DaskExecutor(cluster_address=cluster.scheduler_address)
+
+            success_command = 'echo 1'
+            fail_command = 'exit 1'
+
+            executor.execute_async(key='success', command=success_command)
+            executor.execute_async(key='fail', command=fail_command)
+
+            success_future = next(
+                k for k, v in executor.futures.items() if v == 'success')
+            fail_future = next(
+                k for k, v in executor.futures.items() if v == 'fail')
+
+            time.sleep(0.5)
```

Comment:
I hear you -- here's the full details:

1. The function is being executed in the cluster, so if we proceed immediately it might not have time to complete
2. We could check the future in a `while` loop until it completes, but we'd have to wrap that in a timeout just in case there is an error communicating with the cluster
3. Therefore, my thought was to skip the while loop and just block for a moment to see if the future completes.

However for completeness I'll put that in the code, better to be explicit! One more commit coming.